README
                       E M P I R E  I I

                        version 2.19beta12
                             (Lodi)

                          February, 2004

Compilation:

On UNIX systems just run 'Compile' script inside the main empire 
directory. This will compile all components of the package including
utility codes. If you have used 'setup-emp' script to install 
EMPIRE-II this step is not needed as the whole package should 
already be compiled.

Running:

Use lrun.tcl script in the empire/work directory. This needs Tcl/Tk
and bash shell installed. 

Try also beta version of the new GUI (Xrun.tcl in the work directory).
Although not fully completed, this interface provides all functionalities
of the previous 'lrun.tcl' and much more (e.g., ZVV plotting of double
differential and single spectra, file managment features, and multiple
runs). Xrun.tcl needs Tcl/Tk, incr Widgets, and bash shell. The 
former three are included in most Linux distributions (e.g., RedHat) 
but might not be installed by default. They can also be downloaded free 
of charge from many web sites, e.g.: 

http://tcl.sourceforge.net/ (Tcl/Tk),
http://incrtcl.sourceforge.net/iwidgets/index.html (incr Widgets). 

Otherwise one can use scripts provided in the empire/work directory. 
Please refer to EMPIRE-2.18 write-up in the empire/doc directory for 
more details.


Report problems to: mwherman@bnl.gov





ChangeLog
1.  Bug in print.f fixed (R. Capote)
2.  Correction to the vibrational-rotational input model routine
    preparing ECIS input inside tl.f (H. Wienke)
3.  Vibrational enhancement in level densities and its damping corrected
    (M. Herman)
4.  Mixed (exclusive/inclusive) ENDF-6 reactions classification implemented
    (M. Herman)
5.  Support for NJOY added to GUI (M. Herman)
6.  Random sampling (among 2+,3-,4+) for high lying levels with unknown
    spin implemented. These levels could be used for DWBA calculations
    (R. Capote)
7.  Kalbach parameterization of double differential emission cross section
    for all possible combinations of incident/outgoing particles
    (including gammas) implemented within PCROSS (R. Capote).
8.  Normalization of the gamma-ray strength function (around 1 keV)
    to the experimental 2*pi*Gg/D0 value implemented. Gg and D0 are
    retrieved from the RIPL-2 database (R. Capote)
    (The first energy point in the calculations must be lower than 40 keV)
9.  Experimental s and p strength functions are retrieved from the RIPL-2
    database and compared to calculated values for incident energies
    below 100 keV (R. Capote)
10. Automatic retrieval of collective levels extended to include levels in the
    continuum, using ECDWBA input keyword (R. Capote)
11. DWBA calculated inelastic scattering cross sections for selected levels
    could be added to continuum. Corresponding "discrete levels" are marked
    "cont" in the discrete level file. The inelastic contribution to the
    continuum is spread over several emission energy bins using Gaussian
    distribution with sigma given in input (new input keyword RESOLF)
12. Additional dummy levels can be placed in the discrete level region in the
    collective level file. The inelastic cross section calculated by DBWA
    to this level is added to the nearest discrete level (R. Capote)
13. Detailed printing of the angular distribution for all collective discrete
    levels implemented in long output (R. Capote)
14. Particle-hole LD formula employed in PCROSS taken from RIPL-2 LD chapter.
    Potential depth considered  (R. Capote)
15. EMPEND algorithm for DDXS fitting improved to accurately fit calculated
    EMPIRE results for elastic and inelastic XS (A. Trkov)
16. Separate energy dependent angular grid for elastic scattering implemented,
    allowing to use any number of points for accurate description of angular 
    distributions at energies above 10 MeV (R. Capote)
17. Mixed (exclusive/inclusive) ENDF-6 cross sections and spectra calculations
    implemented (M. Herman). 
18. Unitarity problem for DIRECT<>0 cases identified and corrected. Unitarity
    check introduced (R. Capote).
19. EMPEND was extended to calculate total production cross section, including 
    MT=5 contributions (A. Trkov)
20. PCROSS is given precedence over DEGAS for PE particle emission (M. Herman)
21. Some  systematics from EAF-3.0 included and made available from 
    GUI->Outputs->x-sec systematics (M. Herman)
22. Energy grid in MSD-Tristan increased by factor of 2 to reduce 
    fluctuations (M. Herman)
23. Optional parameters can be placed within incident energies (line must 
    start with $). The change affects only incident energies that come 
    after (M. Herman).
24. If TUNE for gammas in CN is not specified calculations are done with
    0.999 and Gamma_gamma is renormalized to available experimental data; 
    setting TUNE to 1 turns off normalization (M. Herman).
25. ENDF utility codes (CHECKR, FIZCON, STANEF, PSYCHE) updated to the 7.02
    version (M. Herman)
26. Maximum energy for the HRTW calculations can be controlled from the 
    optional input (R. Capote)
26. Automatic fit of Optical Model Parameters added by B. Carlson
27. Printing of l=2 neutron strength function added (M. Divadeenam)
28. Response functions for the 5 lowest l transfers in MSD can be normalized 
    with a single input line RESNOR (no l-value must be specified) (M. Herman)
29. Effect of ATILNO changed! It renormalizes currently used level density 
    parameter a-tilde preserving internal normalization set by the average 
    ratio between systematics and experimental data. Effective renormalization
    factor is ATILNO*<exp/sys> rather than ATILNO (M. Herman)
30. Prompt fission neutron spectrum calculation including post-fission neutrons 
    emitted from fully accelerated fragments (see Kornilov N.V. et al, Physics 
    of Atomic Nuclei ,62 (1999), pp.173-185) and pre-fission neutrons  emitted 
    after neutron or proton emission of the fissioning CN. These neutrons are 
    treated exactly within EMPIRE HF calculation. Implemented algorithm is an 
    extension of the one by Maslov et al. (Eur. Phys. J. A18 (2003) 93-102) 
    and allows to go above 20 MeV up to about 150 MeV. Nu-bar is also an extension 
    of the formalism published by Maslov et al. (Eur. Phys. J. A18 (2003) 93-102) 
    allowing to extend the energy range from 20 MeV up to about 150 MeV.
    Both, nu-bar and PFNS calculations are consistent with the calculated fission 
    cross sections. The implemented model allows for a better description of PFNS 
    above the emissive threshold than the Los Alamos model. (R. Capote & M. Sin)
32. Giant Multipole sum rules in the continuum are implemented (considering the 
    collective states with spin 0,2,3 included in the collective level file). 
    The results are printed in the long output. Eventually, the new listed resonances 
    could be manually added to the collective level file to fulfill corresponding 
    sum rule. If the sum rule is exhausted, then the corresponding deformation  
    is set to zero. The empirical formulae used for estimation of energies 
    and widths of Giant Multipole Resonances were taken from the TALYS manual. 
    No checking is done of the collective strength used in the MSC+MSD calculations.
    (R. Capote)

source/io.h
Ccc   * $Author: herman $ 
Ccc   * $Date: 2008-08-25 06:11:31 $
Ccc   * $Id: io.h,v 1.36 2008-08-25 06:11:31 herman Exp $
            
c     NOTE UNITS OPENED IN MODULES
c
c     Unit#  Module
c
c      1     tl
c      3     MSD-Orion
c      4     ddhms
c      5     io.h
c      6     io.h
c      7     empire_ctl
c      8     MSD-Orion
c      9     ddhms
c     10     ddhms
c     11     io.h
c     12     io.h
c     13     io.h,input
c     14     io.h
c     15     MSD-Orion,MSD-Tristan
c     16     MSD-Tristan
c     17     empire_ctl
c     18     empire_ctl
c     19     input,MSD-Orion
c     20     input,MSD-Orion
c     21     input
c     22     input
c     23     io.h
c     24     io.h
c     25     (seems to be used but don't know where)
c     26     empire-ctrl, main
c     27     ddhms,input
c     28     ddhms
c     29     io.h
c     31     input
c     32     input,tl
c     33     io.h
c     34     lev-dens, empire_ctl 
c     35     lev-dens, empire_ctl
c     36     lev-dens, empire_ctl
c     38     lev-dens
c     39     tl
c     40     main,chi2 
c     41     main (table of x-sections)
c     42     io.h (degas)
c     43     io.h
c     44     empire_ctl (standard input)
c     45     tl,fusion,main
c     46     tl,fusion,main
c     47     fusion,input
c     48     systematics
c     51     input
c     52     input
c     53     fusion
c     54     HF-comp
c     55     ecis03
c     58     ecis03
c     59     ecis03
c     60     ecis03,main
c     61     ecis03
c     62     ecis03,main
c     63     ecis03
c     64     ecis03
c     65     ecis03
c     66     ecis03, MSD-Tristan
c     68     main
c     70     fileprep (empire_ctl fit)
c     71     fileprep (empire_ctl fit)
c     72     fileprep (empire_ctl fit)
c     73     main.f (for pfns OUTPUT)
c     74     main.f (for pfns OUTPUT)
c     75     ecis03
c     76     ecis03
c     77     input
c     78     ecis03
c     79     input
c     80     main
c     81     input,lev-dens
c     82     input,lev-dens
c     83     input
c     84     input
c     85     ecis03
c     86     ecis03
c     87     ecis03
c     88     ecis03
c     89     ecis03
c     90     ecis03
c     91     ecis-03,MSD-Tristan
c     92     empire_ctl
c     93     ecis-03
c     94     input.f,main
c     95     input.f
c     96     tl
c     97     tl
c     98     fission XS 
c     99     ecis03
c
      OPEN (UNIT = 5,FILE='INPUT.DAT', STATUS='OLD')
      OPEN (UNIT = 6,FILE='LIST.DAT' , STATUS='UNKNOWN')
C-----spin distribution from file SDFILE
      OPEN (UNIT = 43,FILE='SDREAD', STATUS='OLD', ERR=768)
      SDREAD=.TRUE.
      GOTO 869
 768  SDREAD=.FALSE.
 869  CONTINUE
C-----fusion transmission coefficients from file FUSION
      OPEN (UNIT = 11,FILE='FUSION'   , STATUS='OLD', ERR=778)
      FUSREAD=.TRUE.
      GOTO 889
 778  FUSREAD=.FALSE.
 889  CONTINUE
      OPEN (UNIT = 12,FILE='OUTPUT.DAT')
      OPEN (UNIT = 13,FILE='LEVELS', STATUS='OLD', ERR=777)
      FILEVEL=.TRUE.
      GOTO 888
 777  FILEVEL=.FALSE.
      OPEN (UNIT = 14, FILE='LEVELS', STATUS='NEW')
 888  CONTINUE
      OPEN (UNIT = 33,FILE='OMPAR.DIR' ,STATUS='OLD', ERR=879)
C     Added to check if file is not empty
      READ(33,*,END=881)
      REWIND(33)
      OMPARfCC=.TRUE.
      GOTO 891
 881  CLOSE(33,STATUS='DELETE')
 879  OMPARfCC=.FALSE.
      OPEN (UNIT = 33, FILE='OMPAR.DIR', STATUS='NEW')
891   CONTINUE
      OPEN (UNIT = 29,FILE='OMPAR.RIPL' ,STATUS='OLD', ERR=780)
C     Added to check if file is not empty
      READ(29,*,END=776)
      REWIND(29)
      OMPAR_RIPLF=.TRUE.
      GOTO 895
 776  CLOSE(29,STATUS='DELETE')
 780  OMPAR_RIPLF=.FALSE.
      OPEN (UNIT = 29, FILE='OMPAR.RIPL', STATUS='NEW')
 895  CONTINUE
      OPEN (UNIT = 23,FILE='../data/nparac.dat'
     *,STATUS='OLD')
      OPEN (UNIT = 24,FILE='../data/ldp.dat'
     *,STATUS='OLD')
      OPEN (UNIT = 26,FILE='../RIPL-2/optical/om-data/om-parameter-u.dat
     *',STATUS='OLD')
C     OPEN (UNIT = 30,FILE='GAMMA.DAT')
C     OPEN (UNIT = 41,FILE='DEGASINPUT',  STATUS = 'UNKNOWN')
C     OPEN (UNIT = 42,FILE='DEGASRESULT', STATUS = 'UNKNOWN')
C     OPEN (UNIT = 53,FILE='LOW_ENERGY.OUT', STATUS = 'UNKNOWN')
C     OPEN (UNIT = 68,FILE='ELASTIC.DAT', STATUS = 'UNKNOWN')  ! for Chris
C     OPEN (UNIT = 95,FILE='COVAR.DAT', STATUS = 'UNKNOWN')      



